include/contractor/contractor.hpp

```diff
@@ -1,1158 +1,73 @@
-#ifndef CONTRACTOR_HPP
-#define CONTRACTOR_HPP
+#ifndef PROCESSING_CHAIN_HPP
+#define PROCESSING_CHAIN_HPP
 
```

Comment:
Shouldn't the header guard reflect the header? I'd love to even include the hierarchy of folders, e.g. `CONTRACTOR_CONTRACTOR_HPP` in this case to make sure that we can handle files of same names in different folders.